### ISSUE TITLE
[bug] Many 'token for conan operation' on Conan 1.

### ISSUE THREAD

**Issue by gsemet** [conan-io/conan#15528]
### Environment details

* Operating System+version: Linux ubuntu 22.04
* Conan version: 1.52

### Logs

We are using Conan 1.52, and since a few days we see many token apparently automatically "token for conan operation" each time a `conan user -p` is executed in our CI, despit we already are using an identity token.

```
$ conan user -p $ARTIFACTORY_IDENTITY_TOKEN -r my-remote $$ARTIFACTORY_USERNAME
```

![image](https://github.com/conan-io/conan/assets/133498/96646c0c-00fd-41aa-ae76-cfcca42041bd)

Why does it do it when we are already using a token ?

Where is it coded? I do not find it anywhere

**Comment by memsharded** [comment]
Hi @gsemet 

It seems that this could be more related to some changes in the server side. Conan client does not have any special management of tokens, it just sends them in the password field. What Artifactory version are you using? Did you upgrade it recently?

**Comment by gsemet** [comment]
We are using Artifactory EnterpriseX license 7.71.5 rev. I'll check with my support team. Thanks for your answer, there is no hidden, magical login code using by Conan...

**Comment by akleber** [comment]
We are seeing the same issue after updating to Artifactory 7.71.8. Unfortunately we only have a "Pro" license which does not entitle us to report bugs. Maybe someone from the Conan community with access to Artifactory support can report this?

**Comment by memsharded** [comment]
@gsemet did you manage to get some answer? do you have some ticket number already? 

I am having a look, but I'd need further details to reproduce:

- I have downloaded latest Artifactory 7.77.5 (CE)
- Using latest Conan 1.X
- Configured a Conan local repo, got a token in the UI
- Repeating ``conan user -p <token> -r=artifactory admin`` multiple times
- Checking ``ui/admin/configuration/security/access_tokens``, I can only see one token

If you can please provide some steps that I could try to reproduce, that would really help.

**Comment by akleber** [comment]
I can provide more info from our installation. Conan 1.63, Artifactory 7.71.8
* Create Identity token (description: "conan test") for user "foo" in UI
* `conan user -p <token conan test> -r=artifactory foo`
* In the Identity token list of the user a new token with description "token for conan operation" appears. (also visible in `ui/admin/configuration/security/access_tokens`)
* After locally deleting `.conan.db` and running the `conan user` command again, a new "token for conan operation" appears. Happens for every repetition of this step.

I also noticed that I cannot revoke the tokens, neither the manually created "conan test" one nor the "token for conan operation" ones. Not sure if this is related.

So I guess this is not helping much with respect to reproducibility as I performed the same steps. I will try to setup a fresh Artifactory and retest.

**Comment by akleber** [comment]
I can confirm that with the same Conan client I used for reproducing the issue with the company Artifactory, I do not see the issue when using a locally running Artifactory 7.77.5 (CE).

**Comment by memsharded** [comment]
This is curious, I opened my local CE, and I was able to see some "token for conan operation", I removed them, trying to reproduce, but I cannot longer create more.

In any case, just some comment about Conan protocol:

- The identity token or password is the initial auth point
- But when ``conan user`` or ``conan remote login`` (2.0) is done, then Conan will use the password or identity token to negotiate a JWT token with the server. For every command of this, or once the JWT is expired, a new auth will be done and a new JWT token will be generated in Artifactory.
- So it is expected that Artifactory will create and store a new JWT token for Conan operation for every auth.
- I think the difference might be in visualization maybe? Like in some cases these tokens are being shown and not in other cases?

**Comment by memsharded** [comment]
I am confused, I cannot reproduce. 
Maybe those tokens are shown in the UI for CE after some time? The other day the tokens were not there, but today they shown on the UI when I launched it.

Still, it is reasonable to have the "token for Conan operation", and I was able to remove them (as admin), this is kind of expected, please let me know if the above explanation makes sense or not. So I am mostly trying to understand now why in some cases they are shown in the UI and why they aren't in other cases.

**Comment by gsemet** [comment]
I agree that this is quite confusing, we have 2 instances of artifactory, with the same version but not the same config (i am not admin so i cannot know the differences). We see many of these jwt token on one but not on the other. On my account sometimes I see many sometimes i do not see them...

but the thing with CI, since the config is not shared between runner, the jwt is recreate each time (from a token so i do not see the point into recreating a token from another token). So we basically have thousands of tokens in the ui, and it slows down a lot the ui...

**Comment by memsharded** [comment]
I have keep trying to reproduce, but no luck :(

I think this mostly belongs to the server side, so as you do have an enterprise license @gsemet I'd suggest to keep reporting via your support channel and pushing regarding this. 

I'll keep trying on my end to see if I can reproduce or find some related hints.

**Comment by memsharded** [comment]
@gsemet if you have some jira or support ticket, please let me know, here, or privately to info@conan.io

**Comment by gsemet** [comment]
i will try, but you may now or corporate world works, i have to find the right person... For the moment, I have written a script to remove all tokens...
we create 150 tokens every day in our CI...

just a question: why create a jwt token on connection ? Is it conan that does it? Why not just using the token given in the -p parameter (that can be a password, in which cas i understand, but when it is already a token...) ?

**Comment by memsharded** [comment]
> just a question: why create a jwt token on connection ? Is it conan that does it? Why not just using the token given in the -p parameter (that can be a password, in which cas i understand, but when it is already a token...) ?

Conan doesn't differentiate between a token or a password, they are both provided in the "password" field of the authentication, and as such, Conan client would never store them, it could be a security risk. Instead of that the client negotiate with the server a temporary JWT and stores it for further connections, it is way less risky than storing a password or an Identity token with higher privileges than the JWT token necessary to communicate with this specific repository.

I have been checking some of the servers we use for ConanCenter, and I can see these tokens "for conan operation", that would be JWT tokens:

- We have a bunch of them, but not so many as hundreds per day
- They have  a relatively short (days) time span. After that time span they are expired and they disappear.
- I think it is possible the default duration is 1 month, after that they should expire and automatically disappear.

So maybe it is just a configuration thing, and all you want is to set the expire time for the tokens like 1 day, so every day they are automatically removed?

**Comment by Wishmaster69200** [comment]
Same issue in my Artifactory 7.71.8.
Conan client version: 2.8.0
problem appears when the client authenticate on the api: /api/conan/<repo_key>/v2/users/authenticate
any news ?
I think i open ticket to jfrog support

**Comment by volodymyrjf** [comment]
Hi,
This behavior is not specific to Conan, but rather Artifactory-specific.
It looks like the description for the token has been updated for clarity in newer Artifactory versions — it’s now named ConanClient[SetMeUp].
This token is automatically generated by Artifactory when a user clicks “Generate Token & Create Instructions.”
So, if users don’t click that button and simply reuse an existing token, no new tokens will be created.
Let me know if you need further clarification!

<img width="958" alt="Image" src="https://github.com/user-attachments/assets/a88ef346-60fb-4ccd-8462-f42dbd1cfd9e" />
